src/fabric.c

```diff
@@ -507,6 +532,10 @@ void DEFAULT_SYMVER_PRE(fi_freeinfo)(struct fi_info *info)
 			free(info->fabric_attr->prov_name);
 			free(info->fabric_attr);
 		}
+		if (info->handle && info->handle->fclass == FI_CLASS_NIC) {
+			struct fid_nic *nic = (struct fid_nic*) info->handle;
```

Comment:
I don't think you need this, you can just directly `ofi_free_nic(info->handle);`.

Reply:
The types are different -- a cast of some flavor or another is necessary.

I chose to make the param of `ofi_free_nic()` be `(struct fid_nic *)` because, well, you're freeing a `struct fid_nic`, not a generic `fid`.  So I figured the onus should be on the caller to make sure the right type is passed.

Reply:
> The types are different -- a cast of some flavor or another is necessary.

Ooops, I thought `handle` here was `void *`.